string name
string path 
bool read
---
bool success